zenoh-jni/src/session.rs

```diff
@@ -347,12 +365,21 @@ pub unsafe extern "C" fn Java_io_zenoh_jni_JNISession_declareSubscriberViaJNI(
     mut env: JNIEnv,
     _class: JClass,
     key_expr_ptr: *const KeyExpr<'static>,
-    ptr: *const zenoh::Session,
+    key_expr_str: JString,
+    session_ptr: *const zenoh::Session,
     callback: JObject,
     on_close: JObject,
     reliability: jint,
 ) -> *const zenoh::subscriber::Subscriber<'static, ()> {
-    match declare_subscriber(&mut env, key_expr_ptr, ptr, callback, on_close, reliability) {
+    match declare_subscriber(
+        &mut env,
+        key_expr_ptr,
+        key_expr_str,
+        session_ptr,
+        callback,
+        on_close,
+        reliability,
+    ) {
         Ok(subscriber_ptr) => subscriber_ptr,
         Err(err) => {
             _ = err.throw_on_jvm(&mut env).map_err(|err| {
```

Comment:
nit: There May be a better pattern to write this, but its low prio.

Reply:
This exception throwing logic could be centralised somewhere. However this would enter the scope of the error handling refactor I mentioned in another comment previously, to do when we have more bandwidth.